compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -914,6 +914,39 @@ public Integer hash(Node node) {
         } else if (node.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
             BLangSimpleVarRef simpleVarRef = (BLangSimpleVarRef) node;
             result = simpleVarRef.variableName.hashCode();
+        } else if (node.getKind() == NodeKind.STRING_TEMPLATE_LITERAL) {
```

Comment:
What will happen if two tables have different containers as keys, but the expressions are same? will there be a key collision? for example one table, string template and other table, list constructor? there can be a key collision as we take exprs only to calculate the hash?